### PR TITLE
PS2 API and XEC driver

### PR DESCRIPTION
**Pull request by franciscomunoz**
Introduce PS2 APIs and sample code which shows mouse configuration and interaction using the board
mec15xxevb_assy6853.

### CODE REVIEW

File: drivers/ps2/Kconfig.xec [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig PS2_XEC
+	bool "XEC Microchip PS2 driver"
+	help
+	  Enable the Microchip XEC PS2 driver.
+
+config PS2_XEC
```
Comment (dcpleung) [inline drivers/ps2/Kconfig.xec]: As this duplicates the same `PS2_XEC` symbol as the menuconfig above, would this work? I think kconfig would complain about this.

File: drivers/ps2/ps2_mchp_xec.c [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+	if (!callback_isr) {
+		return -EINVAL;
+	}
+
+	data->callback_isr = callback_isr;
+	/* In case the self test for a PS2 device already finished and
+	 * set the SOURCE bit to 1 we clear it before enabling the
+	 * interrupts. Instances must be allocated before the BAT or
+	 * the host may my time out
```
Comment (dcpleung) [inline drivers/ps2/ps2_mchp_xec.c]: extra "my"

File: samples/drivers/ps2/README.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
+.. espi-sample:
```
Comment (dcpleung) [inline samples/drivers/ps2/README.rst]: Please replace references to eSPI in this file with PS2.

**Review by dcpleung — CHANGES_REQUESTED** [review]

**Review by albertofloyd — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by dbkinder — CHANGES_REQUESTED** [review]
We need to update the API documentation to include this PS2 API.  I would expect it in here: https://docs.zephyrproject.org/latest/reference/peripherals/index.html 

See doc/reference/peripherals/i2s.rst for an example for creating ps2.rst (and adding it to the peripherals/index.rst ).

Thanks!

**Comment by franciscomunoz** [comment]
@dbkinder Can you please take a look?

File: samples/drivers/ps2/README.rst [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+The sample can be built and executed on boards supporting PS2.
+Please connect a PS2 mouse in order to exercise the functionality.
+
+Sample output
+=============
+
+.. code-block:: console
+
+   PS2 test with mouse.!
```
[the remark this answers is not in the document]
Comment (dbkinder) [reply samples/drivers/ps2/README.rst]: OK!!!!

File: samples/drivers/ps2/README.rst [hunk 5]
```diff
@@ -0,0 +1,26 @@
+.. ps2-sample:
+
+PS2 Interface
+####################################
+
+Overview
+********
+
+This sample demonstrates how to use the PS2 API.
```
Comment (dbkinder) [inline samples/drivers/ps2/README.rst]: Now that we've got it, you can link to it:
```
This sample demonstrates how to use the :ref:`PS/2 API <ps2>`.
```

File: doc/reference/peripherals/ps2.rst [hunk 6]
```diff
@@ -0,0 +1,21 @@
+.. _ps2:
+
+
+PS2
```
Comment (dbkinder) [inline doc/reference/peripherals/ps2.rst]: Technically, the name is ``PS/2`` (while the Playstation 2 is abbreviated PS2) so change mentions of PS2 in the text to be PS/2

File: samples/drivers/ps2/README.rst [hunk 7]
```diff
@@ -0,0 +1,26 @@
+.. ps2-sample:
+
+PS2 Interface
```
Comment (dbkinder) [inline samples/drivers/ps2/README.rst]: Change ``PS2`` to ``PS/2``

File: samples/drivers/ps2/README.rst [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+********
+
+This sample demonstrates how to use the PS2 API.
+Callbacks are registered that will write to the console indicating PS2 events.
+These events indicate mouse configuration responses and mouse interaction.
+
+Building and Running
+********************
+
+The sample can be built and executed on boards supporting PS2.
```
Comment (dbkinder) [inline samples/drivers/ps2/README.rst]: Changes these mentions of ``PS2`` to ``PS/2`` as well.

File: doc/reference/peripherals/ps2.rst [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+####
+
+Overview
+********
+
```
Comment (dbkinder) [inline doc/reference/peripherals/ps2.rst]: How about something like this for the Overview:
```
The PS/2 connector first hit the market in 1987 on
IBM's desktop PC line of the same name before
becoming an industry-wide standard for mouse and
keyboard connections. Starting around 2007, USB
superseded PS/2 and is the modern peripheral device
connection standard. For legacy support on boards
with a PS/2 connector, Zephyr provides these PS/2 driver APIs.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/reference/peripherals/ps2.rst (hunk 9 above)
Comment (franciscomunoz) [reply doc/reference/peripherals/ps2.rst]: NP, I even updated every occurrence of PS2 in the commit messages.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks!

**Pull request merged**